bash_utils/config.sh

```diff
@@ -21,18 +20,28 @@ function set_env {
             export $(cat $envfile | sed 's/#.*//g' | xargs)
         fi
     done
+    # TODO this is a terrible spot for this, docker setup would make more sense but then .env must be loaded
+    mc config host add spaces $AWS_S3_ENDPOINT $AWS_ACCESS_KEY_ID $AWS_SECRET_ACCESS_KEY --api S3v4
+}
+
+
+function set_error_traps {
+  # Exit when any command fails
+  set -e
+  # keep track of the last executed command
+  trap 'last_command=$current_command; current_command=$BASH_COMMAND' DEBUG
+  # echo an error message before exiting
+  trap 'echo "\"${last_command}\" command filed with exit code $?."' EXIT
 }
 
 
 function run_sql_file {
-    local filename = ${1}
-    psql ${BUILD_ENGINE} --set ON_ERROR_STOP=1 --file ${filename}
+    psql ${BUILD_ENGINE} --set ON_ERROR_STOP=1 --single-transaction --file "$@"
```

Comment:
nice. is the `single-transaction` flag useful because files almost always have more than one command and this rolls them back on an error?

Reply:
Exactly. This will be a moot point if we move to dbt but for now this makes sense for developer QOL - can just restart a script at a specific sql file call if something goes wrong